src/ProxyManager/Factory/LazyLoadingGhostFactory.php

```diff
@@ -36,6 +35,35 @@ class LazyLoadingGhostFactory extends AbstractLazyFactory
     private $generator;
 
     /**
+     * @var array|null
+     */
+    private $keepIntactProperties;
+
+    public function createProxy($className, Closure $initializer, array $properties = null)
+    {
+        $this->keepIntactProperties = $properties;
+        $proxyClassName = $this->generateProxy($className);
+
+        return $proxyClassName::staticProxyConstructor($initializer);
+    }
+
+    private function generateProxyClass($proxyClassName, $className, array $proxyParameters)
+    {
+        $className = $this->configuration->getClassNameInflector()->getUserClassName($className);
+        $phpClass  = new ClassGenerator($proxyClassName);
+
+        $this->getGenerator()->generate(new ReflectionClass($className), $phpClass);
+
+        $phpClass = $this->configuration->getClassSignatureGenerator()->addSignature($phpClass, $proxyParameters);
```

Comment:
`$proxyParameters` should contain the `$properties` here, in order to have class names being correctly handled internally